### ISSUE TITLE
clang-format doesn't remove line break after attributes (e.g., [[nodiscard]])

### ISSUE THREAD

**Issue by bernardosulzbach** [llvm/llvm-project#107701]
With clang-format version 18.1.8, a C++ input of

```cpp
[[nodiscard]]
std::array<float, 2> GetTargetSize() const noexcept;
```

does not get formatted to

```cpp
[[nodiscard]] std::array<float, 2> GetTargetSize() const noexcept;
```

even if the `ColumnLimit` allows for it.

This struck me as odd, and it seems to prevent clang-format from ever removing that line break once it decided to insert it, no matter how much the following function signature changes.

Is this intentional? If so, is there a way to instruct clang-format to not preserve those line breaks?

**Comment by owenca** [comment]
Did you try `BreakAfterAttributes: Never`?

**Comment by bernardosulzbach** [comment]
No. I searched for "annotation" when I should have searched for "attribute" in the documentation. I have updated the issue title to account for that. Anyway, I'm impressed "Leave" even exists, and I see similar values for other options seem to exist as well.

I'm not sure how "never" "Never" is. I'd like clang-format to add line breaks to prevent violations of the column limit. For instance, with the artificial limit of 40, clang-format formats

```cpp
[[nodiscard]] VeryVeryVeryVeryVeryLongType f();
```

to

```cpp
[[nodiscard]] VeryVeryVeryVeryVeryLongType
f();
```

which is still above the limit and doesn't have to be.

But I can live with that. Closing as "Never" is good enough.

Thanks.
